appcues/src/main/java/com/appcues/statemachine/Transitions.kt

```diff
@@ -71,7 +71,7 @@ internal interface Transitions {
                     // in different groups we want to wait for StartStep action from AppcuesViewModel
                     val response = CompletableDeferred<ResultOf<State, Error>>()
                     Transition(
-                        state = EndingStep(experience, flatStepIndex, true) {
+                        state = EndingStep(experience, flatStepIndex, nextStepIndex > flatStepIndex) {
```

Comment:
key spots here on 74 and 84 fix the core issue of "moving to a previous step should not mark complete"